modules/router-store/spec/serializers.spec.ts

```diff
@@ -0,0 +1,192 @@
+
+    const expected = {
+      url: 'url',
+      root: {
+        ...createExpectedSnapshot(),
+        firstChild: createExpectedSnapshot('child'),
+        children: [createExpectedSnapshot('child')],
+      },
+    };
+
+    expect(actual).toEqual(expected);
+  });
+
+  function createExpectedSnapshot(prefix = 'root') {
+    return {
+      ...createRouteSnapshot(prefix),
+      component: `${prefix}-route.routeConfig.component`,
+      root: undefined,
+      parent: undefined,
+      firstChild: undefined,
+      pathFromRoot: undefined,
+    };
+  }
+});
+
+describe('minimal serializer', () => {
+  it('should serialize only the minimal properties', () => {
+    const serializer = new MinimalRouterStateSerializer();
+    const snapshot = createRouteSnapshot();
+    const routerState = {
+      url: 'url',
+      root: snapshot,
+    } as RouterStateSnapshot;
+
+    const actual = serializer.serialize(routerState);
+    const expected = {
+      url: 'url',
+      root: createExpectedSnapshot(),
+    };
+    expect(actual).toEqual(expected);
+  });
+
+  it('should serialize with an empty routeConfig', () => {
+    const serializer = new MinimalRouterStateSerializer();
+    const snapshot = { ...createRouteSnapshot(), routeConfig: null };
+    const routerState = {
+      url: 'url',
+      root: snapshot,
+    } as RouterStateSnapshot;
+
+    const actual = serializer.serialize(routerState);
+    const expected = {
+      url: 'url',
+      root: {
+        ...createExpectedSnapshot(),
+        routeConfig: null,
+      },
+    };
+    expect(actual).toEqual(expected);
+  });
+
+  it('should serialize children', () => {
+    const serializer = new MinimalRouterStateSerializer();
+    const snapshot = {
+      ...createRouteSnapshot(),
+      children: [createRouteSnapshot('child')],
+    };
+    const routerState = {
+      url: 'url',
+      root: snapshot,
+    } as RouterStateSnapshot;
+
+    const actual = serializer.serialize(routerState);
+
+    const expected = {
+      url: 'url',
+      root: {
+        ...createExpectedSnapshot(),
+        firstChild: createExpectedSnapshot('child'),
+        children: [createExpectedSnapshot('child')],
+      },
+    };
+
+    expect(actual).toEqual(expected);
+  });
+
+  function createExpectedSnapshot(prefix = 'root') {
+    let snapshot = {
+      ...createRouteSnapshot(prefix),
+      routeConfig: {
+        // config doesn't have a component because it isn't serializable
+        path: `${prefix}-route.routeConfig.path`,
+        pathMatch: `${prefix}-route.routeConfig.pathMatch`,
+        redirectTo: `${prefix}-route.routeConfig.redirectTo`,
+        outlet: `${prefix}-route.routeConfig.outlet`,
+      },
+      firstChild: undefined,
+    };
+
+    // properties that aren't serializable
+    delete snapshot.paramMap;
+    delete snapshot.queryParamMap;
+    delete snapshot.component;
+
+    // properties that do not exist on the minimal serializer
```

Comment:
Is this OK to leave out? 
I did this because these are always `undefined`

Reply:
Yep